hrabsence/CRM/HRAbsence/Form/AbsenceType.php

```diff
@@ -106,11 +106,32 @@ public function buildQuickForm() {
     $this->addFormRule(array('CRM_HRAbsence_Form_AbsenceType', 'formRule'), $this);
   }
 
-  static function formRule($fields, $files, $self) {
-    $errors = array();
+  /**
+   * Used in validation from addFormRule
+   *
+   * @param array $fields
+   *  An array of fields from the form
+   * @param array $files
+   *  An array of uploaded files
+   * @param self $self
+   *  An instance of this class
+   * @return array
+   *   An array of errors
+   */
+  public static function formRule($fields, $files, $self) {
+    $errors = [];
     if (!array_key_exists('allow_debits', $fields) && !array_key_exists('allow_credits', $fields)) {
       $errors['allow_debits'] = $errors['allow_credits'] = ts("Please choose either 'Allow Debits' and/or 'Allow Credits'");
     }
+
+    $title = CRM_Utils_Array::value('title', $fields);
+    if ($self->getAction() === CRM_Core_Action::ADD) {
+      $existing = civicrm_api3('HRAbsenceType', 'get', ['name' => $title]);
```

Comment:
I've tried implementing this and had a working solution. However I noticed if I click on the type name and edit directly on the /civicrm/absence/type page and the request is sent by AJAX the validation is not called. This is really not what I expected here and I think it could cause trouble later on. 

Did I put the validation in the wrong place or do we have somewhere we usually put it so it will always be called?